pkg/provisioner/fss/fss.go

```diff
@@ -28,248 +28,388 @@ import (
 	"github.com/kubernetes-incubator/external-storage/lib/controller"
 	"github.com/oracle/oci-go-sdk/common"
 	"github.com/oracle/oci-go-sdk/core"
-	"github.com/oracle/oci-go-sdk/filestorage"
+	fss "github.com/oracle/oci-go-sdk/filestorage"
 	"github.com/oracle/oci-go-sdk/identity"
 	"github.com/pkg/errors"
 
 	"github.com/oracle/oci-volume-provisioner/pkg/oci/client"
-	"github.com/oracle/oci-volume-provisioner/pkg/oci/instancemeta"
 	"github.com/oracle/oci-volume-provisioner/pkg/provisioner"
 	"github.com/oracle/oci-volume-provisioner/pkg/provisioner/plugin"
+
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 const (
 	ociVolumeID = "volume.beta.kubernetes.io/oci-volume-id"
 	ociExportID = "volume.beta.kubernetes.io/oci-export-id"
-	fsType      = "fsType"
-	// SubnetID is the name of the parameter which holds the target subnet ocid.
-	SubnetID = "subnetId"
+	// AnnotationMountTargetID configures the mount target to use when
+	// provisioning a FSS volume
+	AnnotationMountTargetID = "volume.beta.kubernetes.io/oci-mount-target-id"
+
 	// MntTargetID is the name of the parameter which hold the target mount target ocid.
 	MntTargetID = "mntTargetId"
 )
 
+const (
+	defaultTimeout  = 5 * time.Minute
```

Comment:
Seems fairly high for a timeout on the FSS resources. Would probably assume longer than a minute indicates a critical error.

Reply:
This is how long we wait for resources to transition out of the CREATING state only. I can't see being more tolerant to creation latency biting us but happy to change it.